src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireService.kt

```diff
@@ -0,0 +1,72 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.incidents
+
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.NotFoundException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Questionnaire
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.QuestionnaireQuestion
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.QuestionnaireRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.specification.QuestionnaireSpecification
+
+@Service
+@Transactional
+class QuestionnaireService(
+  private val questionnaireRepository: QuestionnaireRepository,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getQuestionnaire(questionnaireId: Long): QuestionnaireResponse =
+    questionnaireRepository.findByIdOrNull(questionnaireId)?.let {
+      return mapModel(it)
+    } ?: throw NotFoundException("Questionnaire with id=$questionnaireId does not exist")
+
+  fun findIdsByFilter(pageRequest: Pageable, questionnaireFilter: QuestionnaireFilter): Page<QuestionnaireIdResponse> {
+    log.info("Questionnaire Id filter request : $questionnaireFilter with page request $pageRequest")
+    return questionnaireRepository.findAll(QuestionnaireSpecification(questionnaireFilter), pageRequest)
+      .map { QuestionnaireIdResponse(it.id) }
+  }
+
+  private fun mapModel(entity: Questionnaire): QuestionnaireResponse =
```

Comment:
prefer `toQuestionareResponse` as method name of `toDto` - `mapModel` doesn't quite fit what it isn doing

Reply:
ha ha - I did that initially and then changed it